### PR TITLE
8278255: Add more warning text in ReentrantLock and ReentrantReadWriteLock

### PR DESCRIPTION
**Pull request by viktorklang-ora**
This is an attempt to be more clear about recommendations on Lock usage.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 2 [Reviewers](https://openjdk.org/bylaws#reviewer))

### Issue
 * [JDK-8278255](https://bugs.openjdk.org/browse/JDK-8278255): Add more warning text in ReentrantLock and ReentrantReadWriteLock (**Bug** - P4)

### Reviewers
 * [Pavel Rappo](https://openjdk.org/census#prappo) (@pavelrappo - **Reviewer**)
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18974/head:pull/18974` \
`$ git checkout pull/18974`

Update a local copy of the PR: \
`$ git checkout pull/18974` \
`$ git pull https://git.openjdk.org/jdk.git pull/18974/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18974`

View PR using the GUI difftool: \
`$ git pr show -t 18974`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18974.diff">https://git.openjdk.org/jdk/pull/18974.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18974#issuecomment-2079232270)

### CODE REVIEW

**Comment by viktorklang-ora** [comment]
@stuart-marks Would this be better? 🤔

**Comment by DougLea** [comment]
Thanks for finally processing this after we let this good suggestion sit for too long!

File: src/java.base/share/classes/java/util/concurrent/locks/ReentrantReadWriteLock.java [hunk 1]
```diff
@@ -158,7 +159,7 @@
  *         rwl.writeLock().unlock(); // Unlock write, still hold read
  *       }
  *     }
- *
+ *     // Make sure that code which could throw is executed inside the try block
```
Comment (pavelrappo) [inline src/java.base/share/classes/java/util/concurrent/locks/ReentrantReadWriteLock.java:162]: Nit:
```suggestion
 *     // Make sure that code that could throw is executed inside the try block
```

**Comment by viktorklang-ora** [comment]
@pavelrappo Wdyt?

**Comment by pavelrappo** [comment]
> @pavelrappo Wdyt?

I like it, but maybe native speakers who also write clearly could double check that change from _which_ to _that_? In no particular order: @DougLea, @AlanBateman, @stuart-marks.

**Comment by DougLea** [comment]
The grammar rules prefer "that" because it is a restrictive clause, even though "which" might sound better.  See for example: https://owl.purdue.edu/owl/general_writing/grammar/that_vs_which.html

File: src/java.base/share/classes/java/util/concurrent/locks/ReentrantReadWriteLock.java [hunk 2]
```diff
@@ -158,7 +159,7 @@
  *         rwl.writeLock().unlock(); // Unlock write, still hold read
  *       }
  *     }
- *
+ *     // Make sure that code that could throw is executed inside the try block
```
Comment (viktorklang-ora) [inline src/java.base/share/classes/java/util/concurrent/locks/ReentrantReadWriteLock.java:162]: @pavelrappo Or something like:

```suggestion
 *     // Potentially throwing code should be placed inside the try block
```

**Comment by viktorklang-ora** [comment]
@pavelrappo Ok to integrate? 🤔

**Comment by pavelrappo** [comment]
> @pavelrappo Ok to integrate? 🤔

Oh, you shouldn't have waited for me. I think it's fine. Thanks!

**Review by pavelrappo — APPROVED** [review]

**Pull request closed without merging**